src/Raven.Server/Documents/Sharding/Handlers/Processors/Revisions/ShardedRevisionsHandlerProcessorForGetRevisions.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Raven.Server.Documents.Handlers.Processors.Revisions;
+using Raven.Server.Documents.Sharding.Commands;
+using Raven.Server.Documents.Sharding.Operations;
+using Raven.Server.NotificationCenter.Notifications.Details;
+using Raven.Server.ServerWide.Context;
+using Sparrow.Json;
+using Sparrow.Utils;
+
+namespace Raven.Server.Documents.Sharding.Handlers.Processors.Revisions
+{
+    internal class ShardedRevisionsHandlerProcessorForGetRevisions : AbstractRevisionsHandlerProcessorForGetRevisions<ShardedDatabaseRequestHandler, TransactionOperationContext>
+    {
+        public ShardedRevisionsHandlerProcessorForGetRevisions([NotNull] ShardedDatabaseRequestHandler requestHandler) : base(requestHandler, requestHandler.ContextPool)
+        {
+        }
+
+        protected override async ValueTask GetRevisionByChangeVectorAsync(Microsoft.Extensions.Primitives.StringValues changeVectors, bool metadataOnly, CancellationToken token)
+        {
+            using (ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            {
+                var cmd = new ShardedGetRevisionsByChangeVectorsOperation(changeVectors.ToArray(), metadataOnly, context);
+                
+                var res = await RequestHandler.ShardExecutor.ExecuteParallelForAllAsync(cmd, token);
+
+                if (res == null && changeVectors.Count == 1)
+                {
+                    HttpContext.Response.StatusCode = (int)HttpStatusCode.NotFound;
+                    return;
+                }
+
+                var blittable = RequestHandler.GetBoolValueQueryString("blittable", required: false) ?? false;
+                
+                if (blittable)
+                {
+                    WriteRevisionsBlittable(context, res, out long numberOfResults, out long totalDocumentsSizeInBytes);
+                }
+                else
+                {
+                    await WriteRevisionsResult(context, res);
+                }
+            }
+        }
+
+        protected override async ValueTask GetRevisionsAsync(bool metadataOnly, CancellationToken token)
+        {
+            var id = RequestHandler.GetQueryStringValueAndAssertIfSingleAndNotEmpty("id");
+            var before = RequestHandler.GetDateTimeQueryString("before", required: false);
+            var start = RequestHandler.GetStart();
+            var pageSize = RequestHandler.GetPageSize();
```

Comment:
Something is not correct, I see a lot of code duplication between sharded and non-sharded version of this processor. Ping me tomorrow.